go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -547,15 +547,12 @@ func (ts *trafficSwitcher) dropSourceShards(ctx context.Context) error {
 }
 
 func (ts *trafficSwitcher) switchShardReads(ctx context.Context, cells []string, servedTypes []topodatapb.TabletType, direction TrafficSwitchDirection) error {
-	var fromShards, toShards []*topo.ShardInfo
```

Comment:
What changed that we don't need this code anymore? Are we initializing the source and target shards differently ?

Reply:
Nothing changed. It just wasn't correct. When building the traffic switcher we determine the source and target based on the direction. So this was incorrectly doing something equivalent to what we do in the vtctlclient implementation, when we shouldn't be. It resulted in some test failures.